src/Util/Input.cpp

```diff
@@ -57,19 +57,35 @@ void Input::Update() {
     s_CursorPosition.y =
         -(s_CursorPosition.y - static_cast<float>(WINDOW_HEIGHT) / 2);
 
-    s_LBPressed = s_RBPressed = s_MBPressed = s_Scroll = s_MouseMoving = false;
+    s_Scroll = s_MouseMoving = false;
 
     while (SDL_PollEvent(&s_Event) != 0) {
-        s_LBPressed = (s_Event.type == SDL_MOUSEBUTTONDOWN &&
-                       s_Event.button.button == SDL_BUTTON_LEFT) ||
-                      s_LBPressed;
-        s_RBPressed = (s_Event.type == SDL_MOUSEBUTTONDOWN &&
-                       s_Event.button.button == SDL_BUTTON_RIGHT) ||
-                      s_RBPressed;
-        s_MBPressed = (s_Event.type == SDL_MOUSEBUTTONDOWN &&
-                       s_Event.button.button == SDL_BUTTON_MIDDLE) ||
-                      s_MBPressed;
+        if (s_LBPressed) {
+            s_LBPressed = s_Event.type != SDL_MOUSEBUTTONUP ||
+                          s_Event.button.button != SDL_BUTTON_LEFT;
+        } else {
+            s_LBPressed = (s_Event.type == SDL_MOUSEBUTTONDOWN &&
+                           s_Event.button.button == SDL_BUTTON_LEFT);
+        }
```

Comment:
This is making me so confused and triggering my digital logic PTSD
```suggestion
        if (s_Event.type == SDL_MOUSEBUTTONUP
            && s_Event.button.button == SDL_BUTTON_LEFT) {
            s_LBPressed = false;
        }
        if (s_Event.type == SDL_MOUSEBUTTONDOWN
            && s_Event.button.button == SDL_BUTTON_LEFT) {
            s_LBPressed = true;
        }
```